corehq/sql_accessors/sql_templates/check_form_exists.sql

```diff
@@ -0,0 +1,21 @@
+DROP FUNCTION IF EXISTS check_form_exists(TEXT, TEXT);
+
+CREATE FUNCTION check_form_exists(p_form_id TEXT, domain_name TEXT DEFAULT NULL, form_exists OUT BOOLEAN) AS $$
+DECLARE
+    inner_query   TEXT;
+    select_expr   TEXT := 'SELECT 1 FROM form_processor_xforminstancesql WHERE form_id = $1';
+    domain_filter TEXT := ' AND domain = $2';
+    limit_expr    TEXT := ' LIMIT 1';
+    exists        BOOLEAN;
+BEGIN
+    IF $2 <> '' THEN
+        inner_query := select_expr || domain_filter || limit_expr;
+    ELSE
+        inner_query := select_expr || limit_expr;
+    END IF;
+
+    EXECUTE format('SELECT exists(%s)', inner_query)
+        INTO form_exists
+        USING p_form_id, domain_name;
```

Comment:
@millerdev the query planner will re-plan the query each time (see quotes from docs below). I don't think this will be an issue since none of our queries are complex enough to incur significant planning overhead.

"there is no plan caching for commands executed via EXECUTE. Instead, the command is prepared each time the statement is run. Thus the command string can be dynamically created within the function to perform actions on different tables and columns."

and

"EXECUTE will re-plan the command on each execution, generating a plan that is specific to the current parameter values; whereas PL/pgSQL normally creates a generic plan and caches it for re-use. In situations where the best plan depends strongly on the parameter values, EXECUTE can be significantly faster; while when the plan is not sensitive to parameter values, re-planning will be a waste."